components/project-list/project-list.jsx

```diff
@@ -16,28 +16,48 @@ export default React.createClass({
       displayBatchIndex: 1
     };
   },
+  getDefaultProps() {
+    return {
+      params: {
+        featured: null,
+        issue: null,
+        search: null,
+        ids: null
+      }
+    };
+  },
   componentDidMount() {
     this.fetchData(this.props.params);
   },
   componentWillReceiveProps(nextProps) {
+    // Reset state
+    this.resetState();
```

Comment:
@alanmoo 

> That's why I suggest a spinner or something to indicate that it's doing something, just not a message you have to read with disappearing content.

I'm a bit confused about this part. Did you mean in the transition state (user searched for 'web' first and then 'web literacy') we show

```
input box: [ web literacy ]

spinner icon

[ matching result #1 for 'web' ] [ result #2 ] [ result #3 ]
[ matching result #4 for 'web' ] [ result #6 ] [ result #6 ]
...
...

```